redash/handlers/dashboards.py

```diff
@@ -8,6 +11,19 @@
 from redash.handlers.base import BaseResource, get_object_or_404
 
 
+def _save_change(user, dashboard_id, old_dashboard, new_dashboard, change_type):
```

Comment:
Very good point, our initial implementation was indeed a bit messy and not generic at all. 

This has been refactored and completely generalized. The `_save_change` function has been removed from `dashboards.py` and `queries.py`, and instead we now handle this functionality in the Model directly (see `Change.save_change`). This way things are much cleaner and more re-usable.